perception/autoware_camera_streampetr/src/node.cpp

```diff
@@ -0,0 +1,382 @@
+    return;
+  } else if (tdiff > max_camera_time_diff_ || prediction_timestamp < 0.0) {
+    RCLCPP_WARN(
+      get_logger(),
+      "Couldn't sync cameras. Sync difference: %.2f seconds, timelapsed  from start: %.2f seconds",
+      tdiff, prediction_timestamp);
+    network_->wipe_memory();
+    initial_kinematic_state_ = latest_kinematic_state_;
+    data_store_->restart();
+    return;
+  }
+
+  if (multithreading_) data_store_->freeze_updates();
+  std::vector<autoware_perception_msgs::msg::DetectedObject> output_objects;
+  const auto [ego_pose, ego_pose_inv] = get_ego_pose_vector();
+  if (stop_watch_ptr_) stop_watch_ptr_->tic("latency/inference");
+  std::vector<float> forward_time_ms;
+
+  network_->inference_detector(
+    data_store_->get_image_input(), ego_pose, ego_pose_inv, data_store_->get_image_shape(),
+    data_store_->get_camera_info_vector(),
+    get_camera_extrinsics_vector(data_store_->get_camera_link_names()), prediction_timestamp,
+    output_objects, forward_time_ms);
+
+  if (multithreading_) data_store_->unfreeze_updates();
+
+  double inference_time_ms = -1.0;
+  if (stop_watch_ptr_) inference_time_ms = stop_watch_ptr_->toc("latency/inference", true);
+
+  DetectedObjects output_msg;
+  output_msg.objects = output_objects;
+  output_msg.header.frame_id = "base_link";
+  output_msg.header.stamp = stamp;
+  pub_objects_->publish(output_msg);
+
+  if (debug_publisher_ptr_ && stop_watch_ptr_) {
+    debug_publisher_ptr_->publish<autoware_internal_debug_msgs::msg::Float64Stamped>(
+      "latency/total", stop_watch_ptr_->toc("latency/total", true));
+    debug_publisher_ptr_->publish<autoware_internal_debug_msgs::msg::Float64Stamped>(
+      "latency/preprocess", data_store_->get_preprocess_time_ms());
+    debug_publisher_ptr_->publish<autoware_internal_debug_msgs::msg::Float64Stamped>(
+      "latency/inference", inference_time_ms);
+    debug_publisher_ptr_->publish<autoware_internal_debug_msgs::msg::Float64Stamped>(
+      "latency/inference/backbone", forward_time_ms[0]);
+    debug_publisher_ptr_->publish<autoware_internal_debug_msgs::msg::Float64Stamped>(
+      "latency/inference/ptshead", forward_time_ms[1]);
+    debug_publisher_ptr_->publish<autoware_internal_debug_msgs::msg::Float64Stamped>(
+      "latency/inference/pos_embed", forward_time_ms[2]);
+    debug_publisher_ptr_->publish<autoware_internal_debug_msgs::msg::Float64Stamped>(
+      "latency/inference/postprocess", forward_time_ms[3]);
+    debug_publisher_ptr_->publish<autoware_internal_debug_msgs::msg::Float64Stamped>(
+      "latency/cycle_time_ms", stop_watch_ptr_->toc("latency/cycle_time_ms", true));
+    stop_watch_ptr_->tic("latency/cycle_time_ms");
+  }
+}
+
+std::vector<float> StreamPetrNode::get_camera_extrinsics_vector(
+  const std::vector<std::string> & camera_links)
+{
+  std::vector<float> intrinsics_all = data_store_->get_camera_info_vector();
+
+  std::vector<float> res;
+  res.reserve(camera_links.size() * 16);
```

Comment:
[NITS] `16` is the magic number, so we should avoid it and it would be clear to declare as follows:

```cpp
constexpr size_t num_row = 4;
constexpr size_t num_col = 4;
res.reserve(camera_links.size() * num_row * num_col);

// ...
size_t offset = i * num_row * num_col;
for (size_t row = 0; row < num_row; ++row) {
  for (size_t col = 0; col < num_col; ++col) {
    // ...
  }
}
```